### PR TITLE
Created Ball class that contains information about Ball and overrided…

### PR DESCRIPTION
**Pull request by Trayvor**
… method toString(). Created enum Color that contains colors for balls and CollorSupplier class that return a random color. Created Lottery class that use ColorSupplier to generate random color for ball and also generates a random number for ball. In Application class inizializated main method that create and print information about 3 unique balls.

### CODE REVIEW

File: src/main/java/core/basesyntax/Application.java [hunk 1]
```diff
@@ -3,5 +3,13 @@
...
     public static void main(String[] args) {
         // create three balls using class Lottery and print information about them in console
+        Lottery lottery = new Lottery();
+        Ball[] ballArray = { lottery.getRandomBall(),
+                lottery.getRandomBall(),
+                lottery.getRandomBall() };
+
+        for (Ball selectedBall : ballArray) {
+            System.out.println(selectedBall.toString());
+        }
```
Comment (olekskov) [inline src/main/java/core/basesyntax/Application.java:13]: Please combine and perform filling and printing in one cycle

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 2]
```diff
@@ -1,7 +1,10 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
-    public String getRandomColor() {
-        return null;
+    public Color getRandomColor() {
+        int index = new Random().nextInt(Color.values().length);
```
Comment (olekskov) [inline src/main/java/core/basesyntax/ColorSupplier.java:7]: Please extract random as class variable same as done in Lottery class in order not to create new object each time method is called

File: src/main/java/core/basesyntax/Lottery.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    private Random randomNumber = new Random();
+    private ColorSupplier randomColor = new ColorSupplier();
+
+    public Ball getRandomBall() {
+        int ballNumber = randomNumber.nextInt(100);
```
Comment (olekskov) [inline src/main/java/core/basesyntax/Lottery.java:10]: Please extract magic numbers to constants with proper naming

**Review by olekskov — CHANGES_REQUESTED** [review]

**Comment by Trayvor** [comment]
I added everything you suggested. What changes are requested?

File: src/main/java/core/basesyntax/Application.java [hunk 4]
```diff
@@ -3,5 +3,12 @@
...
     public static void main(String[] args) {
         // create three balls using class Lottery and print information about them in console
+        Lottery lottery = new Lottery();
+        Ball[] ballArray = new Ball[3];
+
+        for (int i = 0; i < ballArray.length; i++) {
+            ballArray[i] = lottery.getRandomBall();
+            System.out.println(ballArray[i].toString());
+        }
     }
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Application.java:13]: we don't need an array, call print on the object returned by getRandomBall method rightaway :)
toString is called on object being printed by default, it's redundant

File: src/main/java/core/basesyntax/Lottery.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+public class Lottery {
+    static final int MAX_BALL_NUMBER = 100;
+    private Random randomNumber = new Random();
+    private ColorSupplier randomColor = new ColorSupplier();
+
+    public Ball getRandomBall() {
+        int ballNumber = randomNumber.nextInt(MAX_BALL_NUMBER);
+        Color ballColor = randomColor.getRandomColor();
+        return new Ball(ballColor.toString(), ballNumber);
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Lottery.java:13]:
```suggestion
    static final int MAX_BALL_NUMBER = 100;
    private Random randomNumber = new Random();
    private ColorSupplier randomColor = new ColorSupplier();

    public Ball getRandomBall() {
        int ballNumber = randomNumber.nextInt(MAX_BALL_NUMBER);
        Color ballColor = randomColor.getRandomColor();
        return new Ball(ballColor.name(), ballNumber);
```

File: src/main/java/core/basesyntax/Application.java [hunk 6]
```diff
@@ -3,5 +3,12 @@
 public class Application {
     public static void main(String[] args) {
         // create three balls using class Lottery and print information about them in console
+        Lottery lottery = new Lottery();
+        Ball[] ballArray = new Ball[3];
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Application.java:7]: better extract 3 to a class constant and use it in for loop

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]
Good job

**Pull request closed without merging**